core/src/main/java/com/redhat/lightblue/client/LightblueClient.java

```diff
@@ -17,6 +17,6 @@
 
     LightblueBulkDataResponse bulkData(AbstractDataBulkRequest<AbstractLightblueDataRequest> requests) throws LightblueException;
 
-	<T> T data(AbstractLightblueDataRequest lightblueRequest, Class<T> type) throws LightblueException;
+	<T> T data(LightblueRequest lightblueRequest, Class<T> type) throws LightblueException;
```

Comment:
I don't agree. Then there should be two interfaces, and two methods. Having
an interface, and then requiring the abstract implementation of it is ugly.

On Mon, May 16, 2016 at 7:00 PM, Dennis Crissman <notifications@github.com>
wrote:

> In core/src/main/java/com/redhat/lightblue/client/LightblueClient.java
> <https://github.com/lightblue-platform/lightblue-client/pull/293#discussion_r63451488>
> :
>
> > @@ -17,6 +17,6 @@
> >
> >      LightblueBulkDataResponse bulkData(AbstractDataBulkRequest<AbstractLightblueDataRequest> requests) throws LightblueException;
> >
> > -	<T> T data(AbstractLightblueDataRequest lightblueRequest, Class<T> type) throws LightblueException;
> > +	<T> T data(LightblueRequest lightblueRequest, Class<T> type) throws LightblueException;
>
> I feel strongly that the data api should be locked down to
> AbstractLightblueDataRequest implementations. The method is not designed to
> work with other types of implementations.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-client/pull/293/files/6ba57d62c328c422cf8891eca563e7ca7de8bac8#r63451488>
>